bitnami/mongodb/templates/hidden/statefulset.yaml

```
@@ -84,14 +89,17 @@ spec:
         - name: volume-permissions
           image: {{ include "mongodb.volumePermissions.image" . }}
           imagePullPolicy: {{ .Values.volumePermissions.image.pullPolicy | quote }}
+          env:
+            - name: BITNAMI_DEBUG
+              value: {{ ternary "true" "false" (or .Values.volumePermissions.image.debug .Values.diagnosticMode.enabled) | quote }}
           command:
             - /bin/bash
             - -ec
           args:
             - |
               mkdir -p {{ .Values.hidden.persistence.mountPath }}{{- if .Values.hidden.persistence.subPath }}/{{ .Values.hidden.persistence.subPath }}{{- end }}
-              {{- if and .Values.podSecurityContext.enabled .Values.containerSecurityContext.enabled }}
-              chown -R "{{ .Values.containerSecurityContext.runAsUser }}:{{ .Values.podSecurityContext.fsGroup }}" "{{ .Values.hidden.persistence.mountPath }}{{- if .Values.hidden.persistence.subPath }}/{{ .Values.hidden.persistence.subPath }}{{- end }}"
+              {{- if and .Values.hidden.podSecurityContext.enabled .Values.hidden.containerSecurityContext.enabled }}
+              chown -R "{{ .Values.hidden.containerSecurityContext.runAsUser }}:{{ .Values.hidden.podSecurityContext.fsGroup }}" "{{ .Values.hidden.persistence.mountPath }}{{- if .Values.hidden.persistence.subPath }}/{{ .Values.hidden.persistence.subPath }}{{- end }}"
               {{- end }}
```

To avoid changing ownership of ".snapshot" and "lost+found", I'd suggest:
```suggestion
          command:
            - /bin/bash
          args:
            - -ec
            - |
              mkdir -p {{ printf "%s/%s" .Values.hidden.persistence.mountPath (default "" .Values.hidden.persistence.subPath) }}
              chown {{ .Values.hidden.containerSecurityContext.runAsUser }}:{{ .Values.hidden.podSecurityContext.fsGroup }} {{ printf "%s/%s" .Values.hidden.persistence.mountPath (default "" .Values.hidden.persistence.subPath) }}
              find  {{ printf "%s/%s" .Values.hidden.persistence.mountPath (default "" .Values.hidden.persistence.subPath) }} -mindepth 1 -maxdepth 1 -not -name ".snapshot" -not -name "lost+found" | xargs -r chown -R {{ .Values.hidden.containerSecurityContext.runAsUser }}:{{ .Values.hidden.podSecurityContext.fsGroup }}

---

The above could also be applied to the other Statefulsets

---

Great! Good catch!